src/analysis/convex/basic.lean

```diff
@@ -279,6 +279,18 @@ lemma convex.add_smul_mem (h : convex s) {x y : E} (hx : x ∈ s) (hy : x + y 
   {t : ℝ} (ht : t ∈ Icc (0 : ℝ) 1) : x + t • y ∈ s :=
 by { convert h.add_smul_sub_mem hx hy ht, abel }
 
+lemma convex.smul_mem_of_zero_mem (h : convex s) {x : E} (zero_mem : (0:E) ∈ s) (hx : x ∈ s)
+  {t : ℝ} (ht : t ∈ Icc (0 : ℝ) 1) : t • x ∈ s :=
+by simpa using h.add_smul_mem zero_mem (by simpa using hx) ht
+
+lemma convex.mem_smul_of_zero_mem (h : convex s) {x : E} (zero_mem : (0:E) ∈ s) (hx : x ∈ s)
+  {t : ℝ} (ht : t ∈ Ici (1 : ℝ)) : x ∈ t • s :=
```

Comment:
why not spell `ht` in the simpler (and defeq) form `1 ≤ t`?

Reply:
I don't object, but I guessed that the normal form for this assumption was as `Ici`! I'll amend :)